### ISSUE TITLE
[firebase_storage]: putData() method doesn't contain any metadata on the web. Permission denied when uploading file (User is not authorized to perform the desired action.) 

### ISSUE THREAD

**Issue by hmirza1** [firebase/flutterfire#12607]
### Is there an existing issue for this?

- [X] I have searched the existing issues.

### Which plugins are affected?

Core, Storage

### Which platforms are affected?

Web

### Description

putData() method from firebase_storage doesn't contain any metadata on web. When uploading a zip to firebase storage, I get an error: [firebase_storage/unauthorized] User is not authorized to perform the desired action.

When printing the metadata contentType and customMetadata on the web, the following error is produced (more in the logs below).

`Error: TypeError: null: type 'Null' is not a subtype of type 'String'`

The same code works completely fine on MacOS and the contentType and customMetadata prints without any errors. 

### Dependencies in pubspec.yaml
```
firebase_auth: ^4.19.0
firebase_core: ^2.28.0
cloud_firestore: ^4.16.0
firebase_storage: ^11.7.0
```

### Reproducing the issue

1. Create a new flutter project with the following dependencies.
```
firebase_auth: ^4.19.0
firebase_core: ^2.28.0
cloud_firestore: ^4.16.0
firebase_storage: ^11.7.0
```

2. Initialise the firebase project inside the flutter app, enable auth and log in.

3. Store a file inside a variable as bytes (Uint8List). For our purposes, we are uploading a zip file.

4. Create a storage reference for the file.
```
final storageRef = FirebaseStorage.instance.ref(); 
final zipRef = storageRef.child("storage/$uiud/zip.zip");
```

5. Set metadata
```
final metadata = SettableMetadata(
    contentType: "application/zip",
    customMetadata: {
      'user_id': 'user_uid_here',
    },
  );
```

6. use putData() method.
```
zipRef.putData(zipBytes, metadata).snapshotEvents.listen((taskSnapshot) {
    print('Task snapshot state is --- ${taskSnapshot.state}');
    print('METADATA CHECK --- ${taskSnapshot.metadata?.customMetadata ?? 'NO CUSTOM METADATA'}');
    print('METADATA CHECK --- ${taskSnapshot.metadata?.contentType ?? 'NO CONTENT TYPE METADATA'}');
    switch (taskSnapshot.state) {
      case TaskState.running:
        double progress =
            100.0 * (taskSnapshot.bytesTransferred / taskSnapshot.totalBytes);
        progress.isNaN ? progress = 0 : null;
        print("Upload is $progress% complete.");
        break;
      case TaskState.paused:
        print("Upload is paused.");
        break;
      case TaskState.canceled:
        print("Upload was cancelled");
        break;
      case TaskState.error:
        print('Upload has error');
        break;
      case TaskState.success:
        print('success');
    }
  }, onError: (error) {
    print(error);
    print('Inside onError block');
  });
```

The error log on web essentially shows the below error every time we receive a new taskSnapshot
```
Task snapshot state is --- TaskState.running
Error: TypeError: null: type 'Null' is not a subtype of type 'String'
dart-sdk/lib/_internal/js_dev_runtime/private/ddc_runtime/errors.dart 297:3          throw_
dart-sdk/lib/_internal/js_shared/lib/rti.dart 1385:3                                 _failedAsCheck
dart-sdk/lib/_internal/js_shared/lib/rti.dart 1363:3                                 _generalAsCheckImplementation
dart-sdk/lib/_internal/js_shared/lib/js_util_patch.dart 81:5                         getProperty
packages/firebase_storage_web/src/interop/storage.dart 214:33                        get bucket
packages/firebase_storage_web/src/utils/metadata.dart 16:23                          fbFullMetadataToFullMetadata
packages/firebase_storage_web/src/task_snapshot_web.dart 34:32                       get metadata
packages/firebase_storage/src/task_snapshot.dart 25:43                               get metadata
dart-sdk/lib/async/zone.dart 1594:9                                                  runUnaryGuarded
dart-sdk/lib/async/stream_impl.dart 339:5                                            [_sendData]
dart-sdk/lib/async/stream_impl.dart 271:7                                            [_add]
dart-sdk/lib/async/stream_pipe.dart 123:11                                           [_add]
dart-sdk/lib/async/stream_pipe.dart 218:9                                            [_handleData]
dart-sdk/lib/async/stream_pipe.dart 153:5                                            [_handleData]
dart-sdk/lib/async/zone.dart 1594:9                                                  runUnaryGuarded
dart-sdk/lib/async/stream_impl.dart 339:5                                            [_sendData]
dart-sdk/lib/async/stream_impl.dart 271:7                                            [_add]
dart-sdk/lib/async/stream_pipe.dart 123:11                                           [_add]
dart-sdk/lib/async/stream_pipe.dart 253:9                                            [_handleData]
dart-sdk/lib/async/stream_pipe.dart 153:5                                            [_handleData]
dart-sdk/lib/async/zone.dart 1594:9                                                  runUnaryGuarded
dart-sdk/lib/async/stream_impl.dart 339:5                                            [_sendData]
dart-sdk/lib/async/stream_impl.dart 271:7                                            [_add]
dart-sdk/lib/async/broadcast_stream_controller.dart 377:24                           [_sendData]
dart-sdk/lib/async/broadcast_stream_controller.dart 244:5                            add
dart-sdk/lib/async/zone.dart 1594:9                                                  runUnaryGuarded
dart-sdk/lib/async/stream_impl.dart 339:5                                            [_sendData]
dart-sdk/lib/async/stream_impl.dart 271:7                                            [_add]
dart-sdk/lib/async/stream_pipe.dart 123:11                                           [_add]
dart-sdk/lib/async/stream_pipe.dart 218:9                                            [_handleData]
dart-sdk/lib/async/stream_pipe.dart 153:5                                            [_handleData]
dart-sdk/lib/async/zone.dart 1594:9                                                  runUnaryGuarded
dart-sdk/lib/async/stream_impl.dart 339:5                                            [_sendData]
dart-sdk/lib/async/stream_impl.dart 271:7                                            [_add]
dart-sdk/lib/async/broadcast_stream_controller.dart 377:24                           [_sendData]
dart-sdk/lib/async/broadcast_stream_controller.dart 244:5                            add
packages/firebase_storage_web/src/interop/storage.dart 336:7                         <fn>
dart-sdk/lib/_internal/js_dev_runtime/private/ddc_runtime/operations.dart 550:37     _checkAndCall
dart-sdk/lib/_internal/js_dev_runtime/private/ddc_runtime/operations.dart 555:39     dcall
https://www.gstatic.com/firebasejs/10.8.1/firebase-storage.js 1:26263                <fn>
```
until the upload is progress is 100% - at which point the logs show:
```
[firebase_storage/unauthorized] User is not authorized to perform the desired action.
In ERROR BLOCK
```

For comparison, the logs on MacOS look like this.
```
flutter: Task snapshot state is --- TaskState.running
flutter: METADATA CHECK --- {user_id: user_uid_here}
flutter: METADATA CHECK --- application/zip
flutter: Upload is 7.3667073667073675% complete.
flutter: Task snapshot state is --- TaskState.running
flutter: METADATA CHECK --- {user_id: user_uid_here}
flutter: METADATA CHECK --- application/zip
flutter: Upload is 100.0% complete.
flutter: Task snapshot state is --- TaskState.running
flutter: METADATA CHECK --- {user_id: user_uid_here}
flutter: METADATA CHECK --- application/zip
flutter: Upload is 100.0% complete.
flutter: Task snapshot state is --- TaskState.success
flutter: METADATA CHECK --- {user_id: user_uid_here}
flutter: METADATA CHECK --- application/zip
flutter: success
```

### Firebase Core version

2.28.0

### Flutter Version

3.19.1

### Relevant Log Output

```shell
Task snapshot state is --- TaskState.running
Error: TypeError: null: type 'Null' is not a subtype of type 'String'
dart-sdk/lib/_internal/js_dev_runtime/private/ddc_runtime/errors.dart 297:3          throw_
dart-sdk/lib/_internal/js_shared/lib/rti.dart 1385:3                                 _failedAsCheck
dart-sdk/lib/_internal/js_shared/lib/rti.dart 1363:3                                 _generalAsCheckImplementation
dart-sdk/lib/_internal/js_shared/lib/js_util_patch.dart 81:5                         getProperty
packages/firebase_storage_web/src/interop/storage.dart 214:33                        get bucket
packages/firebase_storage_web/src/utils/metadata.dart 16:23                          fbFullMetadataToFullMetadata
packages/firebase_storage_web/src/task_snapshot_web.dart 34:32                       get metadata
packages/firebase_storage/src/task_snapshot.dart 25:43                               get metadata
dart-sdk/lib/async/zone.dart 1594:9                                                  runUnaryGuarded
dart-sdk/lib/async/stream_impl.dart 339:5                                            [_sendData]
dart-sdk/lib/async/stream_impl.dart 271:7                                            [_add]
dart-sdk/lib/async/stream_pipe.dart 123:11                                           [_add]
dart-sdk/lib/async/stream_pipe.dart 218:9                                            [_handleData]
dart-sdk/lib/async/stream_pipe.dart 153:5                                            [_handleData]
dart-sdk/lib/async/zone.dart 1594:9                                                  runUnaryGuarded
dart-sdk/lib/async/stream_impl.dart 339:5                                            [_sendData]
dart-sdk/lib/async/stream_impl.dart 271:7                                            [_add]
dart-sdk/lib/async/stream_pipe.dart 123:11                                           [_add]
dart-sdk/lib/async/stream_pipe.dart 253:9                                            [_handleData]
dart-sdk/lib/async/stream_pipe.dart 153:5                                            [_handleData]
dart-sdk/lib/async/zone.dart 1594:9                                                  runUnaryGuarded
dart-sdk/lib/async/stream_impl.dart 339:5                                            [_sendData]
dart-sdk/lib/async/stream_impl.dart 271:7                                            [_add]
dart-sdk/lib/async/broadcast_stream_controller.dart 377:24                           [_sendData]
dart-sdk/lib/async/broadcast_stream_controller.dart 244:5                            add
dart-sdk/lib/async/zone.dart 1594:9                                                  runUnaryGuarded
dart-sdk/lib/async/stream_impl.dart 339:5                                            [_sendData]
dart-sdk/lib/async/stream_impl.dart 271:7                                            [_add]
dart-sdk/lib/async/stream_pipe.dart 123:11                                           [_add]
dart-sdk/lib/async/stream_pipe.dart 218:9                                            [_handleData]
dart-sdk/lib/async/stream_pipe.dart 153:5                                            [_handleData]
dart-sdk/lib/async/zone.dart 1594:9                                                  runUnaryGuarded
dart-sdk/lib/async/stream_impl.dart 339:5                                            [_sendData]
dart-sdk/lib/async/stream_impl.dart 271:7                                            [_add]
dart-sdk/lib/async/broadcast_stream_controller.dart 377:24                           [_sendData]
dart-sdk/lib/async/broadcast_stream_controller.dart 244:5                            add
packages/firebase_storage_web/src/interop/storage.dart 336:7                         <fn>
dart-sdk/lib/_internal/js_dev_runtime/private/ddc_runtime/operations.dart 550:37     _checkAndCall
dart-sdk/lib/_internal/js_dev_runtime/private/ddc_runtime/operations.dart 555:39     dcall
https://www.gstatic.com/firebasejs/10.8.1/firebase-storage.js 1:26263                <fn>
Task snapshot state is --- TaskState.running
Error: TypeError: null: type 'Null' is not a subtype of type 'String'
dart-sdk/lib/_internal/js_dev_runtime/private/ddc_runtime/errors.dart 297:3          throw_
dart-sdk/lib/_internal/js_shared/lib/rti.dart 1385:3                                 _failedAsCheck
dart-sdk/lib/_internal/js_shared/lib/rti.dart 1363:3                                 _generalAsCheckImplementation
dart-sdk/lib/_internal/js_shared/lib/js_util_patch.dart 81:5                         getProperty
packages/firebase_storage_web/src/interop/storage.dart 214:33                        get bucket
packages/firebase_storage_web/src/utils/metadata.dart 16:23                          fbFullMetadataToFullMetadata
packages/firebase_storage_web/src/task_snapshot_web.dart 34:32                       get metadata
packages/firebase_storage/src/task_snapshot.dart 25:43                               get metadata
dart-sdk/lib/async/zone.dart 1594:9                                                  runUnaryGuarded
dart-sdk/lib/async/stream_impl.dart 339:5                                            [_sendData]
dart-sdk/lib/async/stream_impl.dart 271:7                                            [_add]
dart-sdk/lib/async/stream_pipe.dart 123:11                                           [_add]
dart-sdk/lib/async/stream_pipe.dart 218:9                                            [_handleData]
dart-sdk/lib/async/stream_pipe.dart 153:5                                            [_handleData]
dart-sdk/lib/async/zone.dart 1594:9                                                  runUnaryGuarded
dart-sdk/lib/async/stream_impl.dart 339:5                                            [_sendData]
dart-sdk/lib/async/stream_impl.dart 271:7                                            [_add]
dart-sdk/lib/async/stream_pipe.dart 123:11                                           [_add]
dart-sdk/lib/async/stream_pipe.dart 253:9                                            [_handleData]
dart-sdk/lib/async/stream_pipe.dart 153:5                                            [_handleData]
dart-sdk/lib/async/zone.dart 1594:9                                                  runUnaryGuarded
dart-sdk/lib/async/stream_impl.dart 339:5                                            [_sendData]
dart-sdk/lib/async/stream_impl.dart 271:7                                            [_add]
dart-sdk/lib/async/broadcast_stream_controller.dart 377:24                           [_sendData]
dart-sdk/lib/async/broadcast_stream_controller.dart 244:5                            add
dart-sdk/lib/async/zone.dart 1594:9                                                  runUnaryGuarded
dart-sdk/lib/async/stream_impl.dart 339:5                                            [_sendData]
dart-sdk/lib/async/stream_impl.dart 271:7                                            [_add]
dart-sdk/lib/async/stream_pipe.dart 123:11                                           [_add]
dart-sdk/lib/async/stream_pipe.dart 218:9                                            [_handleData]
dart-sdk/lib/async/stream_pipe.dart 153:5                                            [_handleData]
dart-sdk/lib/async/zone.dart 1594:9                                                  runUnaryGuarded
dart-sdk/lib/async/stream_impl.dart 339:5                                            [_sendData]
dart-sdk/lib/async/stream_impl.dart 271:7                                            [_add]
dart-sdk/lib/async/broadcast_stream_controller.dart 377:24                           [_sendData]
dart-sdk/lib/async/broadcast_stream_controller.dart 244:5                            add
packages/firebase_storage_web/src/interop/storage.dart 336:7                         <fn>
dart-sdk/lib/_internal/js_dev_runtime/private/ddc_runtime/operations.dart 550:37     _checkAndCall
dart-sdk/lib/_internal/js_dev_runtime/private/ddc_runtime/operations.dart 555:39     dcall
https://www.gstatic.com/firebasejs/10.8.1/firebase-storage.js 1:26263                <fn>
Task snapshot state is --- TaskState.running
Error: TypeError: null: type 'Null' is not a subtype of type 'String'
dart-sdk/lib/_internal/js_dev_runtime/private/ddc_runtime/errors.dart 297:3          throw_
dart-sdk/lib/_internal/js_shared/lib/rti.dart 1385:3                                 _failedAsCheck
dart-sdk/lib/_internal/js_shared/lib/rti.dart 1363:3                                 _generalAsCheckImplementation
dart-sdk/lib/_internal/js_shared/lib/js_util_patch.dart 81:5                         getProperty
packages/firebase_storage_web/src/interop/storage.dart 214:33                        get bucket
packages/firebase_storage_web/src/utils/metadata.dart 16:23                          fbFullMetadataToFullMetadata
packages/firebase_storage_web/src/task_snapshot_web.dart 34:32                       get metadata
packages/firebase_storage/src/task_snapshot.dart 25:43                               get metadata
dart-sdk/lib/async/zone.dart 1594:9                                                  runUnaryGuarded
dart-sdk/lib/async/stream_impl.dart 339:5                                            [_sendData]
dart-sdk/lib/async/stream_impl.dart 271:7                                            [_add]
dart-sdk/lib/async/stream_pipe.dart 123:11                                           [_add]
dart-sdk/lib/async/stream_pipe.dart 218:9                                            [_handleData]
dart-sdk/lib/async/stream_pipe.dart 153:5                                            [_handleData]
dart-sdk/lib/async/zone.dart 1594:9                                                  runUnaryGuarded
dart-sdk/lib/async/stream_impl.dart 339:5                                            [_sendData]
dart-sdk/lib/async/stream_impl.dart 271:7                                            [_add]
dart-sdk/lib/async/stream_pipe.dart 123:11                                           [_add]
dart-sdk/lib/async/stream_pipe.dart 253:9                                            [_handleData]
dart-sdk/lib/async/stream_pipe.dart 153:5                                            [_handleData]
dart-sdk/lib/async/zone.dart 1594:9                                                  runUnaryGuarded
dart-sdk/lib/async/stream_impl.dart 339:5                                            [_sendData]
dart-sdk/lib/async/stream_impl.dart 271:7                                            [_add]
dart-sdk/lib/async/broadcast_stream_controller.dart 377:24                           [_sendData]
dart-sdk/lib/async/broadcast_stream_controller.dart 244:5                            add
dart-sdk/lib/async/zone.dart 1594:9                                                  runUnaryGuarded
dart-sdk/lib/async/stream_impl.dart 339:5                                            [_sendData]
dart-sdk/lib/async/stream_impl.dart 271:7                                            [_add]
dart-sdk/lib/async/stream_pipe.dart 123:11                                           [_add]
dart-sdk/lib/async/stream_pipe.dart 218:9                                            [_handleData]
dart-sdk/lib/async/stream_pipe.dart 153:5                                            [_handleData]
dart-sdk/lib/async/zone.dart 1594:9                                                  runUnaryGuarded
dart-sdk/lib/async/stream_impl.dart 339:5                                            [_sendData]
dart-sdk/lib/async/stream_impl.dart 271:7                                            [_add]
dart-sdk/lib/async/broadcast_stream_controller.dart 377:24                           [_sendData]
dart-sdk/lib/async/broadcast_stream_controller.dart 244:5                            add
packages/firebase_storage_web/src/interop/storage.dart 336:7                         <fn>
dart-sdk/lib/_internal/js_dev_runtime/private/ddc_runtime/operations.dart 550:37     _checkAndCall
dart-sdk/lib/_internal/js_dev_runtime/private/ddc_runtime/operations.dart 555:39     dcall
https://www.gstatic.com/firebasejs/10.8.1/firebase-storage.js 1:26263                <fn>
Task snapshot state is --- TaskState.running
Error: TypeError: null: type 'Null' is not a subtype of type 'String'
dart-sdk/lib/_internal/js_dev_runtime/private/ddc_runtime/errors.dart 297:3          throw_
dart-sdk/lib/_internal/js_shared/lib/rti.dart 1385:3                                 _failedAsCheck
dart-sdk/lib/_internal/js_shared/lib/rti.dart 1363:3                                 _generalAsCheckImplementation
dart-sdk/lib/_internal/js_shared/lib/js_util_patch.dart 81:5                         getProperty
packages/firebase_storage_web/src/interop/storage.dart 214:33                        get bucket
packages/firebase_storage_web/src/utils/metadata.dart 16:23                          fbFullMetadataToFullMetadata
packages/firebase_storage_web/src/task_snapshot_web.dart 34:32                       get metadata
packages/firebase_storage/src/task_snapshot.dart 25:43                               get metadata
dart-sdk/lib/async/zone.dart 1594:9                                                  runUnaryGuarded
dart-sdk/lib/async/stream_impl.dart 339:5                                            [_sendData]
dart-sdk/lib/async/stream_impl.dart 271:7                                            [_add]
dart-sdk/lib/async/stream_pipe.dart 123:11                                           [_add]
dart-sdk/lib/async/stream_pipe.dart 218:9                                            [_handleData]
dart-sdk/lib/async/stream_pipe.dart 153:5                                            [_handleData]
dart-sdk/lib/async/zone.dart 1594:9                                                  runUnaryGuarded
dart-sdk/lib/async/stream_impl.dart 339:5                                            [_sendData]
dart-sdk/lib/async/stream_impl.dart 271:7                                            [_add]
dart-sdk/lib/async/stream_pipe.dart 123:11                                           [_add]
dart-sdk/lib/async/stream_pipe.dart 253:9                                            [_handleData]
dart-sdk/lib/async/stream_pipe.dart 153:5                                            [_handleData]
dart-sdk/lib/async/zone.dart 1594:9                                                  runUnaryGuarded
dart-sdk/lib/async/stream_impl.dart 339:5                                            [_sendData]
dart-sdk/lib/async/stream_impl.dart 271:7                                            [_add]
dart-sdk/lib/async/broadcast_stream_controller.dart 377:24                           [_sendData]
dart-sdk/lib/async/broadcast_stream_controller.dart 244:5                            add
dart-sdk/lib/async/zone.dart 1594:9                                                  runUnaryGuarded
dart-sdk/lib/async/stream_impl.dart 339:5                                            [_sendData]
dart-sdk/lib/async/stream_impl.dart 271:7                                            [_add]
dart-sdk/lib/async/stream_pipe.dart 123:11                                           [_add]
dart-sdk/lib/async/stream_pipe.dart 218:9                                            [_handleData]
dart-sdk/lib/async/stream_pipe.dart 153:5                                            [_handleData]
dart-sdk/lib/async/zone.dart 1594:9                                                  runUnaryGuarded
dart-sdk/lib/async/stream_impl.dart 339:5                                            [_sendData]
dart-sdk/lib/async/stream_impl.dart 271:7                                            [_add]
dart-sdk/lib/async/broadcast_stream_controller.dart 377:24                           [_sendData]
dart-sdk/lib/async/broadcast_stream_controller.dart 244:5                            add
packages/firebase_storage_web/src/interop/storage.dart 336:7                         <fn>
dart-sdk/lib/_internal/js_dev_runtime/private/ddc_runtime/operations.dart 550:37     _checkAndCall
dart-sdk/lib/_internal/js_dev_runtime/private/ddc_runtime/operations.dart 555:39     dcall
https://www.gstatic.com/firebasejs/10.8.1/firebase-storage.js 1:26263                <fn>
Task snapshot state is --- TaskState.running
Error: TypeError: null: type 'Null' is not a subtype of type 'String'
dart-sdk/lib/_internal/js_dev_runtime/private/ddc_runtime/errors.dart 297:3          throw_
dart-sdk/lib/_internal/js_shared/lib/rti.dart 1385:3                                 _failedAsCheck
dart-sdk/lib/_internal/js_shared/lib/rti.dart 1363:3                                 _generalAsCheckImplementation
dart-sdk/lib/_internal/js_shared/lib/js_util_patch.dart 81:5                         getProperty
packages/firebase_storage_web/src/interop/storage.dart 214:33                        get bucket
packages/firebase_storage_web/src/utils/metadata.dart 16:23                          fbFullMetadataToFullMetadata
packages/firebase_storage_web/src/task_snapshot_web.dart 34:32                       get metadata
packages/firebase_storage/src/task_snapshot.dart 25:43                               get metadata
dart-sdk/lib/async/zone.dart 1594:9                                                  runUnaryGuarded
dart-sdk/lib/async/stream_impl.dart 339:5                                            [_sendData]
dart-sdk/lib/async/stream_impl.dart 271:7                                            [_add]
dart-sdk/lib/async/stream_pipe.dart 123:11                                           [_add]
dart-sdk/lib/async/stream_pipe.dart 218:9                                            [_handleData]
dart-sdk/lib/async/stream_pipe.dart 153:5                                            [_handleData]
dart-sdk/lib/async/zone.dart 1594:9                                                  runUnaryGuarded
dart-sdk/lib/async/stream_impl.dart 339:5                                            [_sendData]
dart-sdk/lib/async/stream_impl.dart 271:7                                            [_add]
dart-sdk/lib/async/stream_pipe.dart 123:11                                           [_add]
dart-sdk/lib/async/stream_pipe.dart 253:9                                            [_handleData]
dart-sdk/lib/async/stream_pipe.dart 153:5                                            [_handleData]
dart-sdk/lib/async/zone.dart 1594:9                                                  runUnaryGuarded
dart-sdk/lib/async/stream_impl.dart 339:5                                            [_sendData]
dart-sdk/lib/async/stream_impl.dart 271:7                                            [_add]
dart-sdk/lib/async/broadcast_stream_controller.dart 377:24                           [_sendData]
dart-sdk/lib/async/broadcast_stream_controller.dart 244:5                            add
dart-sdk/lib/async/zone.dart 1594:9                                                  runUnaryGuarded
dart-sdk/lib/async/stream_impl.dart 339:5                                            [_sendData]
dart-sdk/lib/async/stream_impl.dart 271:7                                            [_add]
dart-sdk/lib/async/stream_pipe.dart 123:11                                           [_add]
dart-sdk/lib/async/stream_pipe.dart 218:9                                            [_handleData]
dart-sdk/lib/async/stream_pipe.dart 153:5                                            [_handleData]
dart-sdk/lib/async/zone.dart 1594:9                                                  runUnaryGuarded
dart-sdk/lib/async/stream_impl.dart 339:5                                            [_sendData]
dart-sdk/lib/async/stream_impl.dart 271:7                                            [_add]
dart-sdk/lib/async/broadcast_stream_controller.dart 377:24                           [_sendData]
dart-sdk/lib/async/broadcast_stream_controller.dart 244:5                            add
packages/firebase_storage_web/src/interop/storage.dart 336:7                         <fn>
dart-sdk/lib/_internal/js_dev_runtime/private/ddc_runtime/operations.dart 550:37     _checkAndCall
dart-sdk/lib/_internal/js_dev_runtime/private/ddc_runtime/operations.dart 555:39     dcall
https://www.gstatic.com/firebasejs/10.8.1/firebase-storage.js 1:26263                <fn>
Task snapshot state is --- TaskState.running
Error: TypeError: null: type 'Null' is not a subtype of type 'String'
dart-sdk/lib/_internal/js_dev_runtime/private/ddc_runtime/errors.dart 297:3          throw_
dart-sdk/lib/_internal/js_shared/lib/rti.dart 1385:3                                 _failedAsCheck
dart-sdk/lib/_internal/js_shared/lib/rti.dart 1363:3                                 _generalAsCheckImplementation
dart-sdk/lib/_internal/js_shared/lib/js_util_patch.dart 81:5                         getProperty
packages/firebase_storage_web/src/interop/storage.dart 214:33                        get bucket
packages/firebase_storage_web/src/utils/metadata.dart 16:23                          fbFullMetadataToFullMetadata
packages/firebase_storage_web/src/task_snapshot_web.dart 34:32                       get metadata
packages/firebase_storage/src/task_snapshot.dart 25:43                               get metadata
dart-sdk/lib/async/zone.dart 1594:9                                                  runUnaryGuarded
dart-sdk/lib/async/stream_impl.dart 339:5                                            [_sendData]
dart-sdk/lib/async/stream_impl.dart 271:7                                            [_add]
dart-sdk/lib/async/stream_pipe.dart 123:11                                           [_add]
dart-sdk/lib/async/stream_pipe.dart 218:9                                            [_handleData]
dart-sdk/lib/async/stream_pipe.dart 153:5                                            [_handleData]
dart-sdk/lib/async/zone.dart 1594:9                                                  runUnaryGuarded
dart-sdk/lib/async/stream_impl.dart 339:5                                            [_sendData]
dart-sdk/lib/async/stream_impl.dart 271:7                                            [_add]
dart-sdk/lib/async/stream_pipe.dart 123:11                                           [_add]
dart-sdk/lib/async/stream_pipe.dart 253:9                                            [_handleData]
dart-sdk/lib/async/stream_pipe.dart 153:5                                            [_handleData]
dart-sdk/lib/async/zone.dart 1594:9                                                  runUnaryGuarded
dart-sdk/lib/async/stream_impl.dart 339:5                                            [_sendData]
dart-sdk/lib/async/stream_impl.dart 271:7                                            [_add]
dart-sdk/lib/async/broadcast_stream_controller.dart 377:24                           [_sendData]
dart-sdk/lib/async/broadcast_stream_controller.dart 244:5                            add
dart-sdk/lib/async/zone.dart 1594:9                                                  runUnaryGuarded
dart-sdk/lib/async/stream_impl.dart 339:5                                            [_sendData]
dart-sdk/lib/async/stream_impl.dart 271:7                                            [_add]
dart-sdk/lib/async/stream_pipe.dart 123:11                                           [_add]
dart-sdk/lib/async/stream_pipe.dart 218:9                                            [_handleData]
dart-sdk/lib/async/stream_pipe.dart 153:5                                            [_handleData]
dart-sdk/lib/async/zone.dart 1594:9                                                  runUnaryGuarded
dart-sdk/lib/async/stream_impl.dart 339:5                                            [_sendData]
dart-sdk/lib/async/stream_impl.dart 271:7                                            [_add]
dart-sdk/lib/async/broadcast_stream_controller.dart 377:24                           [_sendData]
dart-sdk/lib/async/broadcast_stream_controller.dart 244:5                            add
packages/firebase_storage_web/src/interop/storage.dart 336:7                         <fn>
dart-sdk/lib/_internal/js_dev_runtime/private/ddc_runtime/operations.dart 550:37     _checkAndCall
dart-sdk/lib/_internal/js_dev_runtime/private/ddc_runtime/operations.dart 555:39     dcall
https://www.gstatic.com/firebasejs/10.8.1/firebase-storage.js 1:26263                <fn>
Task snapshot state is --- TaskState.running
Error: TypeError: null: type 'Null' is not a subtype of type 'String'
dart-sdk/lib/_internal/js_dev_runtime/private/ddc_runtime/errors.dart 297:3          throw_
dart-sdk/lib/_internal/js_shared/lib/rti.dart 1385:3                                 _failedAsCheck
dart-sdk/lib/_internal/js_shared/lib/rti.dart 1363:3                                 _generalAsCheckImplementation
dart-sdk/lib/_internal/js_shared/lib/js_util_patch.dart 81:5                         getProperty
packages/firebase_storage_web/src/interop/storage.dart 214:33                        get bucket
packages/firebase_storage_web/src/utils/metadata.dart 16:23                          fbFullMetadataToFullMetadata
packages/firebase_storage_web/src/task_snapshot_web.dart 34:32                       get metadata
packages/firebase_storage/src/task_snapshot.dart 25:43                               get metadata
dart-sdk/lib/async/zone.dart 1594:9                                                  runUnaryGuarded
dart-sdk/lib/async/stream_impl.dart 339:5                                            [_sendData]
dart-sdk/lib/async/stream_impl.dart 271:7                                            [_add]
dart-sdk/lib/async/stream_pipe.dart 123:11                                           [_add]
dart-sdk/lib/async/stream_pipe.dart 218:9                                            [_handleData]
dart-sdk/lib/async/stream_pipe.dart 153:5                                            [_handleData]
dart-sdk/lib/async/zone.dart 1594:9                                                  runUnaryGuarded
dart-sdk/lib/async/stream_impl.dart 339:5                                            [_sendData]
dart-sdk/lib/async/stream_impl.dart 271:7                                            [_add]
dart-sdk/lib/async/stream_pipe.dart 123:11                                           [_add]
dart-sdk/lib/async/stream_pipe.dart 253:9                                            [_handleData]
dart-sdk/lib/async/stream_pipe.dart 153:5                                            [_handleData]
dart-sdk/lib/async/zone.dart 1594:9                                                  runUnaryGuarded
dart-sdk/lib/async/stream_impl.dart 339:5                                            [_sendData]
dart-sdk/lib/async/stream_impl.dart 271:7                                            [_add]
dart-sdk/lib/async/broadcast_stream_controller.dart 377:24                           [_sendData]
dart-sdk/lib/async/broadcast_stream_controller.dart 244:5                            add
dart-sdk/lib/async/zone.dart 1594:9                                                  runUnaryGuarded
dart-sdk/lib/async/stream_impl.dart 339:5                                            [_sendData]
dart-sdk/lib/async/stream_impl.dart 271:7                                            [_add]
dart-sdk/lib/async/stream_pipe.dart 123:11                                           [_add]
dart-sdk/lib/async/stream_pipe.dart 218:9                                            [_handleData]
dart-sdk/lib/async/stream_pipe.dart 153:5                                            [_handleData]
dart-sdk/lib/async/zone.dart 1594:9                                                  runUnaryGuarded
dart-sdk/lib/async/stream_impl.dart 339:5                                            [_sendData]
dart-sdk/lib/async/stream_impl.dart 271:7                                            [_add]
dart-sdk/lib/async/broadcast_stream_controller.dart 377:24                           [_sendData]
dart-sdk/lib/async/broadcast_stream_controller.dart 244:5                            add
packages/firebase_storage_web/src/interop/storage.dart 336:7                         <fn>
dart-sdk/lib/_internal/js_dev_runtime/private/ddc_runtime/operations.dart 550:37     _checkAndCall
dart-sdk/lib/_internal/js_dev_runtime/private/ddc_runtime/operations.dart 555:39     dcall
https://www.gstatic.com/firebasejs/10.8.1/firebase-storage.js 1:26263                <fn>
[firebase_storage/unauthorized] User is not authorized to perform the desired action.
In ERROR BLOCK
```

### Flutter dependencies

<details>
<summary>Expand <code>Flutter dependencies</code> snippet</summary>
<br>

```yaml

Dart SDK 3.3.0
Flutter SDK 3.19.1
studio 1.0.0+1

dependencies:
- archive 3.4.10 [crypto path pointycastle]
- audio_video_progress_bar 2.0.2 [flutter]
- cloud_firestore 4.16.0 [cloud_firestore_platform_interface cloud_firestore_web collection firebase_core firebase_core_platform_interface flutter meta]
- cupertino_icons 1.0.6
- easy_image_viewer 1.4.1 [flutter]
- equatable 2.0.5 [collection meta]
- file_picker 6.2.1 [flutter flutter_web_plugins flutter_plugin_android_lifecycle plugin_platform_interface ffi path win32]
- file_saver 0.2.12 [flutter flutter_web_plugins path_provider path_provider_windows path_provider_linux dio]
- firebase_auth 4.19.0 [firebase_auth_platform_interface firebase_auth_web firebase_core firebase_core_platform_interface flutter meta]
- firebase_core 2.28.0 [firebase_core_platform_interface firebase_core_web flutter meta]
- firebase_storage 11.7.0 [firebase_core firebase_core_platform_interface firebase_storage_platform_interface firebase_storage_web flutter]
- flutter 0.0.0 [characters collection material_color_utilities meta vector_math sky_engine]
- flutter_animate 4.5.0 [flutter flutter_shaders]
- flutter_hooks 0.20.5 [flutter]
- flutter_octicons 0.42.0 [flutter]
- flutter_portal 1.1.4 [collection flutter vector_math]
- flutter_side_menu 0.4.0 [flutter auto_size_text badges]
- flutter_sticky_header 0.6.5 [flutter value_layout_builder]
- gap 3.0.1 [flutter]
- hooks_riverpod 2.5.1 [collection flutter flutter_hooks flutter_riverpod riverpod state_notifier]
- http 1.2.1 [async http_parser meta web]
- just_audio 0.9.37 [just_audio_platform_interface just_audio_web audio_session rxdart path path_provider async uuid crypto meta flutter]
- loading_animation_widget 1.2.1 [flutter]
- media_kit 1.1.10+1 [meta path image uri_parser collection synchronized uuid http safe_local_storage universal_platform js]
- media_kit_libs_video 1.0.4 [media_kit_native_event_loop media_kit_libs_android_video media_kit_libs_ios_video media_kit_libs_macos_video media_kit_libs_windows_video media_kit_libs_linux]
- media_kit_video 1.2.4 [flutter media_kit synchronized wakelock_plus screen_brightness volume_controller universal_platform plugin_platform_interface]
- mime 1.0.5
- pad 1.0.0 [flutter]
- path 1.9.0
- path_provider 2.1.2 [flutter path_provider_android path_provider_foundation path_provider_linux path_provider_platform_interface path_provider_windows]
- phosphor_flutter 2.0.1 [flutter]
- shadcn_ui 0.2.4 [boxy flutter flutter_animate flutter_localizations flutter_svg rive vector_graphics]
- sliver_tools 0.2.12 [flutter]
- universal_html 2.2.4 [async csslib charcode collection html meta source_span typed_data universal_io]
- url_strategy 0.2.0 [flutter_web_plugins]
- uuid 4.3.3 [crypto sprintf meta fixnum]

dev dependencies:
- flutter_lints 2.0.3 [lints]
- flutter_test 0.0.0 [flutter test_api matcher path fake_async clock stack_trace vector_math leak_tracker_flutter_testing async boolean_selector characters collection leak_tracker leak_tracker_testing material_color_utilities meta source_span stream_channel string_scanner term_glyph vm_service]

transitive dependencies:
- _flutterfire_internals 1.3.28 [collection firebase_core firebase_core_platform_interface flutter meta]
- args 2.4.2
- async 2.11.0 [collection meta]
- audio_session 0.1.19 [flutter flutter_web_plugins rxdart meta]
- auto_size_text 3.0.0 [flutter]
- badges 3.1.2 [flutter]
- boolean_selector 2.1.1 [source_span string_scanner]
- boxy 2.2.0 [flutter vector_math]
- characters 1.3.0
- charcode 1.3.1
- clock 1.1.1
- cloud_firestore_platform_interface 6.1.12 [_flutterfire_internals collection firebase_core flutter meta plugin_platform_interface]
- cloud_firestore_web 3.11.0 [_flutterfire_internals cloud_firestore_platform_interface collection firebase_core firebase_core_web flutter flutter_web_plugins]
- collection 1.18.0
- convert 3.1.1 [typed_data]
- crypto 3.0.3 [typed_data]
- csslib 1.0.0 [source_span]
- dbus 0.7.10 [args ffi meta xml]
- dio 5.4.2+1 [async http_parser meta path]
- fake_async 1.3.1 [clock collection]
- ffi 2.1.2
- firebase_auth_platform_interface 7.2.1 [_flutterfire_internals collection firebase_core flutter meta plugin_platform_interface]
- firebase_auth_web 5.11.0 [firebase_auth_platform_interface firebase_core firebase_core_web flutter flutter_web_plugins http_parser meta web]
- firebase_core_platform_interface 5.0.0 [collection flutter flutter_test meta plugin_platform_interface]
- firebase_core_web 2.14.0 [firebase_core_platform_interface flutter flutter_web_plugins meta web]
- firebase_storage_platform_interface 5.1.15 [_flutterfire_internals collection firebase_core flutter meta plugin_platform_interface]
- firebase_storage_web 3.9.0 [_flutterfire_internals async firebase_core firebase_core_web firebase_storage_platform_interface flutter flutter_web_plugins http meta web]
- fixnum 1.1.0
- flutter_localizations 0.0.0 [flutter intl characters clock collection material_color_utilities meta path vector_math]
- flutter_plugin_android_lifecycle 2.0.17 [flutter]
- flutter_riverpod 2.5.1 [collection flutter meta riverpod state_notifier]
- flutter_shaders 0.1.2 [flutter vector_math]
- flutter_svg 2.0.10+1 [flutter http vector_graphics vector_graphics_codec vector_graphics_compiler]
- flutter_web_plugins 0.0.0 [flutter characters collection material_color_utilities meta vector_math]
- graphs 2.3.1 [collection]
- html 0.15.4 [csslib source_span]
- http_parser 4.0.2 [collection source_span string_scanner typed_data]
- image 4.1.7 [archive meta xml]
- intl 0.18.1 [clock meta path]
- js 0.6.7 [meta]
- just_audio_platform_interface 4.2.2 [flutter plugin_platform_interface]
- just_audio_web 0.4.9 [just_audio_platform_interface flutter flutter_web_plugins]
- leak_tracker 10.0.0 [clock collection meta path vm_service]
- leak_tracker_flutter_testing 2.0.1 [flutter leak_tracker leak_tracker_testing matcher meta]
- leak_tracker_testing 2.0.1 [leak_tracker matcher meta]
- lints 2.1.1
- matcher 0.12.16+1 [async meta stack_trace term_glyph test_api]
- material_color_utilities 0.8.0 [collection]
- media_kit_libs_android_video 1.3.6 [flutter plugin_platform_interface]
- media_kit_libs_ios_video 1.1.4 [flutter plugin_platform_interface]
- media_kit_libs_linux 1.1.3 [flutter]
- media_kit_libs_macos_video 1.1.4 [flutter plugin_platform_interface]
- media_kit_libs_windows_video 1.0.9 [flutter]
- media_kit_native_event_loop 1.0.8 [flutter]
- meta 1.11.0
- package_info_plus 6.0.0 [ffi flutter flutter_web_plugins http meta path package_info_plus_platform_interface web win32]
- package_info_plus_platform_interface 2.0.1 [flutter meta plugin_platform_interface]
- path_parsing 1.0.1 [vector_math meta]
- path_provider_android 2.2.2 [flutter path_provider_platform_interface]
- path_provider_foundation 2.3.2 [flutter path_provider_platform_interface]
- path_provider_linux 2.2.1 [ffi flutter path path_provider_platform_interface xdg_directories]
- path_provider_platform_interface 2.1.2 [flutter platform plugin_platform_interface]
- path_provider_windows 2.2.1 [ffi flutter path path_provider_platform_interface win32]
- petitparser 6.0.2 [meta]
- platform 3.1.4
- plugin_platform_interface 2.1.8 [meta]
- pointycastle 3.8.0 [collection convert js]
- rive 0.13.1 [collection flutter flutter_web_plugins http meta plugin_platform_interface rive_common]
- rive_common 0.3.3 [collection ffi flutter flutter_web_plugins graphs http meta plugin_platform_interface]
- riverpod 2.5.1 [collection meta stack_trace state_notifier]
- rxdart 0.27.7
- safe_local_storage 1.0.2 [path synchronized]
- screen_brightness 0.2.2+1 [flutter screen_brightness_platform_interface screen_brightness_android screen_brightness_ios screen_brightness_macos screen_brightness_windows]
- screen_brightness_android 0.1.0+2 [flutter screen_brightness_platform_interface]
- screen_brightness_ios 0.1.0 [flutter screen_brightness_platform_interface]
- screen_brightness_macos 0.1.0+1 [flutter screen_brightness_platform_interface]
- screen_brightness_platform_interface 0.1.0 [flutter plugin_platform_interface]
- screen_brightness_windows 0.1.3 [flutter screen_brightness_platform_interface]
- sky_engine 0.0.99
- source_span 1.10.0 [collection path term_glyph]
- sprintf 7.0.0
- stack_trace 1.11.1 [path]
- state_notifier 1.0.0 [meta]
- stream_channel 2.1.2 [async]
- string_scanner 1.2.0 [source_span]
- synchronized 3.1.0+1
- term_glyph 1.2.1
- test_api 0.6.1 [async boolean_selector collection meta source_span stack_trace stream_channel string_scanner term_glyph]
- typed_data 1.3.2 [collection]
- universal_io 2.2.2 [collection meta typed_data]
- universal_platform 1.0.0+1
- uri_parser 2.0.2 [path safe_local_storage]
- value_layout_builder 0.3.1 [flutter]
- vector_graphics 1.1.11+1 [flutter http vector_graphics_codec]
- vector_graphics_codec 1.1.11+1
- vector_graphics_compiler 1.1.11+1 [args meta path_parsing xml vector_graphics_codec path]
- vector_math 2.1.4
- vm_service 13.0.0
- volume_controller 2.0.7 [flutter]
- wakelock_plus 1.2.2 [flutter flutter_web_plugins meta wakelock_plus_platform_interface win32 dbus package_info_plus web]
- wakelock_plus_platform_interface 1.2.1 [flutter plugin_platform_interface meta]
- web 0.5.1
- win32 5.4.0 [ffi]
- xdg_directories 1.0.4 [meta path]
- xml 6.5.0 [collection meta petitparser]
```

</details>

### Additional context and comments

Have already ran flutter clean. I can see there is also another issue open at the moment (#12570). I'm not sure if this is the same cause or not, but it may be relevant.

When removing the security rules, we do reach ```TaskState.success``` and the metadata check is printed, but only once taskSnapshot.state == TaskState.success.

Here are the logs from the penultimate and final taskSnapshots:
### taskSnapshot - penultimate 
```
Task snapshot state is --- TaskState.running
Error: TypeError: null: type 'Null' is not a subtype of type 'String'
dart-sdk/lib/_internal/js_dev_runtime/private/ddc_runtime/errors.dart 297:3          throw_
dart-sdk/lib/_internal/js_shared/lib/rti.dart 1385:3                                 _failedAsCheck
dart-sdk/lib/_internal/js_shared/lib/rti.dart 1363:3                                 _generalAsCheckImplementation
dart-sdk/lib/_internal/js_shared/lib/js_util_patch.dart 81:5                         getProperty
packages/firebase_storage_web/src/interop/storage.dart 214:33                        get bucket
packages/firebase_storage_web/src/utils/metadata.dart 16:23                          fbFullMetadataToFullMetadata
packages/firebase_storage_web/src/task_snapshot_web.dart 34:32                       get metadata
packages/firebase_storage/src/task_snapshot.dart 25:43                               get metadata
packages/dojo_studio/features/download/functions/upload_scenario_server.dart 188:43  <fn>
dart-sdk/lib/async/zone.dart 1594:9                                                  runUnaryGuarded
dart-sdk/lib/async/stream_impl.dart 339:5                                            [_sendData]
dart-sdk/lib/async/stream_impl.dart 271:7                                            [_add]
dart-sdk/lib/async/stream_pipe.dart 123:11                                           [_add]
dart-sdk/lib/async/stream_pipe.dart 218:9                                            [_handleData]
dart-sdk/lib/async/stream_pipe.dart 153:5                                            [_handleData]
dart-sdk/lib/async/zone.dart 1594:9                                                  runUnaryGuarded
dart-sdk/lib/async/stream_impl.dart 339:5                                            [_sendData]
dart-sdk/lib/async/stream_impl.dart 271:7                                            [_add]
dart-sdk/lib/async/stream_pipe.dart 123:11                                           [_add]
dart-sdk/lib/async/stream_pipe.dart 253:9                                            [_handleData]
dart-sdk/lib/async/stream_pipe.dart 153:5                                            [_handleData]
dart-sdk/lib/async/zone.dart 1594:9                                                  runUnaryGuarded
dart-sdk/lib/async/stream_impl.dart 339:5                                            [_sendData]
dart-sdk/lib/async/stream_impl.dart 271:7                                            [_add]
dart-sdk/lib/async/broadcast_stream_controller.dart 377:24                           [_sendData]
dart-sdk/lib/async/broadcast_stream_controller.dart 244:5                            add
dart-sdk/lib/async/zone.dart 1594:9                                                  runUnaryGuarded
dart-sdk/lib/async/stream_impl.dart 339:5                                            [_sendData]
dart-sdk/lib/async/stream_impl.dart 271:7                                            [_add]
dart-sdk/lib/async/stream_pipe.dart 123:11                                           [_add]
dart-sdk/lib/async/stream_pipe.dart 218:9                                            [_handleData]
dart-sdk/lib/async/stream_pipe.dart 153:5                                            [_handleData]
dart-sdk/lib/async/zone.dart 1594:9                                                  runUnaryGuarded
dart-sdk/lib/async/stream_impl.dart 339:5                                            [_sendData]
dart-sdk/lib/async/stream_impl.dart 271:7                                            [_add]
dart-sdk/lib/async/broadcast_stream_controller.dart 377:24                           [_sendData]
dart-sdk/lib/async/broadcast_stream_controller.dart 244:5                            add
packages/firebase_storage_web/src/interop/storage.dart 336:7                         <fn>
dart-sdk/lib/_internal/js_dev_runtime/private/ddc_runtime/operations.dart 550:37     _checkAndCall
dart-sdk/lib/_internal/js_dev_runtime/private/ddc_runtime/operations.dart 555:39     dcall
https://www.gstatic.com/firebasejs/10.8.1/firebase-storage.js 1:26263                <fn>
```
### taskSnapshot - final (success with no security rules)
```
Task snapshot state is --- TaskState.success
METADATA CHECK --- {}
METADATA CHECK --- application/zip
success
```

Any help would be appreciated. Many thanks!
